src/main/java/com/nvidia/spark/rapids/jni/BloomFilter.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.jni;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ai.rapids.cudf.ColumnVector;
+import ai.rapids.cudf.ColumnView;
+import ai.rapids.cudf.Cuda;
+import ai.rapids.cudf.CudfException;
+import ai.rapids.cudf.DeviceMemoryBuffer;
+import ai.rapids.cudf.MemoryCleaner;
+import ai.rapids.cudf.NativeDepsLoader;
+
+public class BloomFilter implements AutoCloseable {
+  private static final Logger log = LoggerFactory.getLogger(BloomFilter.class);
+  private final int numHashes;
+  private final long bloomFilterBits;
+  private DeviceMemoryBuffer bloomFilter;
+
+  static {
+    NativeDepsLoader.loadNativeDeps();
+  }
+
+  /**
+   * Construct an empty BloomFilter which uses the specified number of hashes and bits.
+   * @param numHashes The number of hashes to use when adding values and probing.
+   * @param bloomFilterBits The size of the bloom filter in bits.
+   */
+  public BloomFilter(int numHashes, long bloomFilterBits){
+    if(numHashes <= 0){
+      throw new IllegalArgumentException("Bloom filters must have a positive hash count");
+    }
+    if(bloomFilterBits <= 0){
+      throw new IllegalArgumentException("Bloom filters must have a positive number of bits");
+    }
+
+    this.numHashes = numHashes;    
+    this.bloomFilterBits = bloomFilterBits;
+    
+    long bloomFilterBytes = bloomFilterByteSize(bloomFilterBits);
+    bloomFilter = DeviceMemoryBuffer.allocate(bloomFilterBytes);
+    Cuda.asyncMemset(bloomFilter.getAddress(), (byte)0, bloomFilterBytes);
+  }  
+
+  /**
+   * Construct a BloomFilter from a pre-existing buffer, using the specified number of
+   * hashes and bits.
+   * @param numHashes The number of hashes to use when adding values and probing.
+   * @param bloomFilterBits The size of the bloom filter in bits.
+   * @param buffer The pre-existing buffer of.
+   */
+  public BloomFilter(int numHashes, long bloomFilterBits, DeviceMemoryBuffer buffer){
```

Comment:
I think this should take a BaseDeviceMemoryBuffer instead.  That way it can take a zero-copy slice of another buffer rather than making the caller copy it first to get a DeviceMemoryBuffer.